File: libs/core/src/symbols/assembly_symbol.cpp
/**
 * @file assembly_symbol.cpp
 * @author Francesco Corso
 * @date 8/8/2026
 * @brief
 */
module prism.core:symbols.assembly_symbol.impl;

import :symbols.assembly_symbol;

namespace prism
{
    void AssemblySymbol::write_display_string(TextWriter &writer) const
    {
        writer.write(name().as_string_view());
    }
} // namespace prism

File: src/Prism.Core/Symbols/AssemblySymbol.cs
namespace Prism.Core.Symbols;

public abstract class AssemblySymbol : Symbol
{
    private protected AssemblySymbol(string name)
        : base(name) { }

    public sealed override AssemblySymbol? ContainingAssembly => null;

    public sealed override NamespaceSymbol? ContainingNamespace => null;

    public sealed override TypeSymbol? ContainingType => null;

    public abstract NamespaceSymbol GlobalNamespace { get; }

    public sealed override void WriteDisplayString(TextWriter writer)
    {
        writer.Write(Name);
    }
}
